tests/test_mongodb_incremental_open_transactions.py

```diff
@@ -0,0 +1,258 @@
+
+def generate_simple_coll_docs(num_docs):
+    docs = []
+    for int_value in range(num_docs):
+        docs.append({"int_field": int_value, "string_field": random_string_generator()})
+    return docs
+
+class MongoDBOpenTransactions(unittest.TestCase):
+
+    def expected_check_streams_sync_1(self):
+        return {
+            'simple_db-simple_coll_1',
+            'simple_db-simple_coll_2',
+            'simple_db-simple_coll_3'
+        }
+
+    def expected_check_streams_sync_2(self):
+        return {
+            'simple_db-simple_coll_1',
+            'simple_db-simple_coll_2',
+            'simple_db-simple_coll_3'
+        }
+
+    def expected_pks_1(self):
+        return {
+            'simple_db_simple_coll_1': {'_id'},
+            'simple_db_simple_coll_2': {'_id'},
+            'simple_db_simple_coll_3': {'_id'}
+        }
+
+    def expected_pks_2(self):
+        return {
+            'simple_db_simple_coll_1': {'_id'},
+            'simple_db_simple_coll_2': {'_id'},
+            'simple_db_simple_coll_3': {'_id'}
+        }
+
+    def expected_row_counts_sync_1(self):
+        return {
+            'simple_db_simple_coll_1': 10,
+            'simple_db_simple_coll_2': 20,
+            'simple_db_simple_coll_3': 0
+        }
+
+    def expected_row_counts_sync_2(self):
+        return {
+            'simple_db_simple_coll_1': 2,
+            'simple_db_simple_coll_2': 2,
+            'simple_db_simple_coll_3': 5
+        }
+
+    def expected_row_counts_sync_3(self):
+        return {
+            'simple_db_simple_coll_1': 0,
+            'simple_db_simple_coll_2': 0,
+            'simple_db_simple_coll_3': 0
+        }
+
+    def expected_sync_streams_1(self):
+        return {
+            'simple_db_simple_coll_1',
+            'simple_db_simple_coll_2',
+            'simple_db_simple_coll_3'
+        }
+
+    def expected_sync_streams_2(self):
+        return {
+            'simple_db_simple_coll_1',
+            'simple_db_simple_coll_2',
+            'simple_db_simple_coll_3'
+        }
+
+    def name(self):
+        return "tap_tester_mongodb_open_transaction"
+
+    def tap_name(self):
+        return "tap-mongodb"
+
+    def get_type(self):
+        return "platform.mongodb"
+
+    def get_credentials(self):
+        return {'password': os.getenv('TAP_MONGODB_PASSWORD')}
+
+    def get_properties(self):
+        return {'host' : os.getenv('TAP_MONGODB_HOST'),
+                'port' : os.getenv('TAP_MONGODB_PORT'),
+                'user' : os.getenv('TAP_MONGODB_USER'),
+                'database' : os.getenv('TAP_MONGODB_DBNAME'),
+                'include_schemas_in_destination_stream_name': 'true'
+        }
+
+    def test_run(self):
+
+        ensure_environment_variables_set()
+
+        with get_test_connection() as client:
+            # drop all dbs/collections
+            drop_all_collections(client)
+
+
+
+            #################
+            # Session 1
+            #################
```

Comment:
Can you add a docstring somewhere or multiple comments to clearly state how each session corresponds to the test cases.

Reply:
done